src/shell.js

```diff
@@ -230,11 +230,17 @@ if (ENVIRONMENT_IS_SHELL) {
     };
   }
 
-  if (typeof print !== 'undefined') {
-    // Prefer to use print/printErr where they exist, as they usually work better.
-    if (typeof console === 'undefined') console = {};
-    console.log = print;
-    console.warn = console.error = typeof printErr !== 'undefined' ? printErr : print;
+  // Add polyfill for console object for JS shells that don't have this builtin (jsc)
+  if (typeof console === 'undefined') {
+    console = { log: print };
```

Comment:
This doesn't check if `print` exists, but I guess that's ok since if `console` doesn't, `print` is our only hope. Please add a comment though.